modules/python-modules/Makefile.am

```diff
@@ -0,0 +1,125 @@
+	modules/python-modules/syslogng/debuggercli/lexertoken.py		\
+	modules/python-modules/syslogng/debuggercli/macrocompleter.py		\
+	modules/python-modules/syslogng/debuggercli/readline.py			\
+	modules/python-modules/syslogng/debuggercli/syslognginternals.py	\
+	modules/python-modules/syslogng/debuggercli/tablexer.py			\
+	modules/python-modules/syslogng/debuggercli/templatelang.py		\
+	modules/python-modules/syslogng/debuggercli/templatelexer.py		\
+	modules/python-modules/syslogng/debuggercli/tflang.py			\
+	modules/python-modules/syslogng/debuggercli/tests/__init__.py		\
+	modules/python-modules/syslogng/debuggercli/tests/test_completer.py    	\
+	modules/python-modules/syslogng/debuggercli/tests/test_commandlinelexer.py\
+	modules/python-modules/syslogng/debuggercli/tests/test_completerlang.py	\
+	modules/python-modules/syslogng/debuggercli/tests/test_choicecompleter.py\
+	modules/python-modules/syslogng/debuggercli/tests/test_debuglang.py	\
+	modules/python-modules/syslogng/debuggercli/tests/test_debuggercli.py	\
+	modules/python-modules/syslogng/debuggercli/tests/test_getoptlexer.py	\
+	modules/python-modules/syslogng/debuggercli/tests/test_langcompleter.py	\
+	modules/python-modules/syslogng/debuggercli/tests/test_lexer.py		\
+	modules/python-modules/syslogng/debuggercli/tests/test_macrocompleter.py\
+	modules/python-modules/syslogng/debuggercli/tests/test_tablexer.py	\
+	modules/python-modules/syslogng/debuggercli/tests/test_templatelexer.py	\
+	modules/python-modules/syslogng/debuggercli/tests/test_templatelang.py	\
+	modules/python-modules/syslogng/debuggercli/tests/test_tflang.py	\
+	modules/python-modules/pylintrc						\
+	modules/python-modules/test_pymodules.sh				\
+	modules/python-modules/CMakeLists.txt					\
+	\
+	modules/python-modules/syslogng/modules/example/scl/example.conf	\
+	modules/python-modules/syslogng/modules/example/__init__.py		\
+	modules/python-modules/setup.py						\
+	modules/python-modules/README.md
+
+if ENABLE_PYTHON
+
+PYMODULES_PATH = modules/python-modules
+PYMODULES_BUILDDIR = $(abs_builddir)/$(PYMODULES_PATH)
+PYMODULES_SRCDIR = $(top_srcdir)/modules/python-modules
+PYMODULES_MANIFEST = $(PYMODULES_BUILDDIR)/install-manifest.txt
+PYMODULES_ROOT_MODULE = syslogng
+PYTHON_ROOT = $(if $(DESTDIR),$(DESTDIR),/)
+
+INSTALL_EXEC_HOOKS += pymodules-install
+UNINSTALL_HOOKS += pymodules-uninstall
+CLEAN_HOOKS += pymodules-clean
+
+PYMODULES_SETUP_OPTIONS ?= --root="$(PYTHON_ROOT)" --install-lib="$(python_moduledir)"
+IS_PYTEST_INSTALLED := $(shell ${PYTHON} -m pytest --version > /dev/null 2>&1 ; echo $$?)
+
+
+#
+# The syslogng package is installed into ${prefix}/lib/syslog-ng, while its
+# requirements all installed to the normal Python site-packages directory.
+#
+# Therefore we need to separate the installation of both.
+#
+pymodules-install: pymodules-install-requirements
+	(cd $(PYMODULES_SRCDIR) && $(PYTHON) setup.py \
+		build --build-base="$(PYMODULES_BUILDDIR)/build" \
```

Comment:
Shouldn't `build` be during build (`make`) instead of install (`make install`)? Building/installing with cmake does that and it seems cleaner.

Reply:
We don't really need the build step, and install does that automatically. The reason it is there explicitly is that we want to change the default build directory and this is the only way to do that.